File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int zero = 0;
        
        for(int i=0; i<nums.size(); i++){
            if(nums[i] == 0) zero++;
            else if(zero > 0){
                nums[i-zero] = nums[i];
            }
        }
        
        for(int i=nums.size()-zero; i<nums.size(); i++){
            nums[i] = 0;
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        lastNonZero = 0
        
        for i in range(len(nums)):
            if(nums[i] != 0):
                nums[lastNonZero] = nums[i]
                lastNonZero += 1
        
        for i in range(lastNonZero, len(nums)):
            nums[i] = 0
        